### PR TITLE
feat: make chain list extendable

### PR DESCRIPTION
**Pull request by re2005**
The available network list is essential on every application, but they are so many, so better reduce it to a few and let the user extend it as it needs.

Please share ideas here.

### CODE REVIEW

**Comment by chnejohnson** [comment]
It looks good, seems that you just remain `mainnet` and `goerli` as the default networks, so other networks should be added by users, right?

**Comment by re2005** [comment]
It's just a suggestion, maybe this plugin shouldn't take the responsibility of managing the networks, also because there are too many and on each project they will be different.

But also the information, per chain that devs needs will be different for their needs.

So you can let users to just input their networks/chains and don't deal with that.

**Comment by chnejohnson** [comment]
I like this extendable feature, but I prefer to remain `hardhat`, `Arbitrum`, `Rinkarby`, and `Optimism`

**Comment by tempe-techie** [comment]
Yes, this makes sense. I can help with the docs if @re2005 can't. The best would be to have a `.js` or `.json` template file that has some of the most common custom networks already added (Polygon, Optimism, Arbitrum, BSC), so that devs can see how to add new networks.

It makes sense to have Ethereum and its testnets (Goerli) added by default in the plugin (because they use `wallet_switchEthereumChain` to switch). And the custom ones (added by devs) would then use `wallet_addEthereumChain`. Which makes the implementation simpler.

**Comment by re2005** [comment]
Something other lib do provide a list of most common chains. So you can import everything or just the ones you need.

Check:
https://github.com/wagmi-dev/wagmi/blob/main/packages/core/src/constants/chains.ts

it's all in there.

**Comment by re2005** [comment]
Hey @chnejohnson So finally i'm running on a valid/good use case.

My application supports some networks that are not "inside" `vue-dapp` so when I try to connect to some of those, and the network is not yet on `Metamask` than `vue-dapp` looks to it's internal list of networks in order to add it, but because the new network it is not there it will fail with an error. The user at this point can't add a new network to metamask using `vue-dapp`

Check:
https://github.com/chnejohnson/vue-dapp/blob/main/src/wallets/metaMask.ts#L192

This Pr will also fix that allowing the extension of the network list.

I've added some documentation as well. (@tempe-techie please help out if you think documentation needs more details)

Please review and let's merge it.

**Comment by tempe-techie** [comment]
> I've added some documentation as well. (@tempe-techie please help out if you think documentation needs more details)
> 
> Please review and let's merge it.

Looks good! Thanks! 🎉 🤘 😎

File: src/constants/chainId.ts [hunk 1]
```diff
@@ -1,165 +1,36 @@
...
-  },
-  [ChainId.Rinkarby]: {
-    chainId: '0x' + ChainId.Rinkarby.toString(16),
-    chainName: 'RinkArby',
+export const NETWORK_DETAILS: { [key: number]: AddEthereumChainParameter } = {
+  [ChainId.Mainnet]: {
+    chainId: '0x' + ChainId.Mainnet.toString(16),
+    chainName: 'Mainnet',
     rpcUrls: ['https://rinkeby.arbitrum.io/rpc'],
     blockExplorerUrls: ['https://rinkeby-explorer.arbitrum.io'],
```
Comment (chnejohnson) [inline src/constants/chainId.ts:19]: These two lines have some problems

File: src/constants/chainId.ts [hunk 2]
```diff
@@ -1,165 +1,36 @@
...
-    },
-  },
-  [ChainId.Polygon]: {
-    chainId: '0x' + ChainId.Polygon.toString(16),
-    chainName: 'Polygon Mainnet',
+
+  [ChainId.Goerli]: {
+    chainId: '0x' + ChainId.Goerli.toString(16),
+    chainName: 'Goerli',
     rpcUrls: ['https://polygon-rpc.com/'],
```
Comment (chnejohnson) [inline src/constants/chainId.ts:29]: This line is incorrect

File: src/constants/chainId.ts [hunk 3]
```diff
@@ -1,165 +1,36 @@
...
-    blockExplorerUrls: ['https://polygonscan.com/'],
+    blockExplorerUrls: ['https://goerli.etherscan.io/'],
     nativeCurrency: {
       symbol: 'MATIC',
```
Comment (chnejohnson) [inline src/constants/chainId.ts:32]: *ETH

**Review by chnejohnson — CHANGES_REQUESTED** [review]
Hey @re2005, thanks for your good work!

I reviewed the code and found a few bugs (probably made by my hand), but it's fine to fix it later. The main problem is that you have to drop the last commit:`d74597`( Merge branch main...) and run `git rebase -i main` on commit:`ecf088` and fix a simple conflict. So it will be more clear in the git history.

In addition to what's mentioned above, it looks good to me 👍

**Comment by re2005** [comment]
@chnejohnson Thanks for the review 🙏 

I hope It's all fixed now.

**Comment by chnejohnson** [comment]
@re2005 Perfect!

**Pull request merged**